packages/kolibri-common/components/SyncSchedule/EditDeviceSyncSchedule.vue

```diff
@@ -270,40 +275,44 @@
           });
       },
       handleSaveSchedule() {
-        FacilityResource.fetchModel({ id: this.$store.getters.activeFacilityId, force: true }).then(
-          facility => {
-            this.facility = { ...facility };
-            TaskResource.startTask({
-              type: TaskTypes.SYNCPEERFULL,
-              facility: this.facility.id,
-              device_id: this.device.id,
-              baseurl: this.baseurl,
-              enqueue_args: { enqueue_at: this.serverTime, repeat_interval: 2, repeat: 2 },
+        FacilityResource.fetchModel({ id: this.facilityId, force: true }).then(facility => {
+          this.facility = { ...facility };
+          const date = new Date(this.serverTime);
+          const equeue_param = date.toISOString();
+          TaskResource.startTask({
+            type: TaskTypes.SYNCPEERFULL,
+            facility: this.facility.id,
+            device_id: this.$route.params.deviceId,
+            baseurl: this.baseurl,
+            enqueue_args: {
+              enqueue_at: equeue_param,
+              repeat_interval: this.selectedItem.value,
+              repeat: 2,
+            },
+          })
+            .then(() => {
+              history.back();
+              this.showSnackbarNotification('syncAdded');
             })
-              .then(() => {
-                history.back();
-                this.showSnackbarNotification('syncAdded');
-              })
-              .catch(() => {
-                this.createTaskFailedSnackbar();
-              });
-          }
-        );
+            .catch(() => {
+              this.createTaskFailedSnackbar();
+            });
+        });
       },
 
       cancelBtn() {
-        this.$router.push({ name: PageNames.ManageSyncSchedule });
+        this.$router.push({ name: SyncPageNames.MANAGE_SYNC_SCHEDULE });
```

Comment:
We should mix in the `facility_id` parameter from the current route here, or this will break on the facility plugin when we are editing sync schedules for the non-default facility, and all the time for the device plugin where the facility_id is required.